packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,93 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+import { EVENTS } from './utils/const'
+import { setupEvents, removeEvent } from './utils/mapEventHelpers'
+
+@themed(['Gmap_FreeDrawTootip'], { pure: true })
+export default class FreeDrawLayer extends PureComponent {
+  static propTypes ={
+    map: PropTypes.object,
+    onMapDrawStart: PropTypes.func,
+    onMapDrawEnd: PropTypes.func,
+    theme: PropTypes.object,
+  }
+
+  static defaultProps = {
+    theme: {},
+    map: {},
+  }
+
+  enableMapControls() {
+    const { map } = this.props
+    map.setOptions({
+      draggable: true,
+      zoomControl: true,
+      scrollwheel: true,
+      disableDoubleClickZoom: true,
+    })
```

Comment:
What happens if the map had `zoomControl:false` to start out with? Then after user completes freedraw, `zoomControl` will be turned on.

Perhaps before you disable those options, you could save the current state in `this.previousMapOptions` then restore the same values when freedraw is complete.

Reply:
Good point :+1: